test/ml/algorithms/test_qkernel.py

```diff
@@ -0,0 +1,160 @@
+                                                     seed_transpiler=self.random_seed)
+
+        self.feature_map = ZZFeatureMap(feature_dimension=2, reps=2)
+
+        self.sample_train = np.asarray([[2.95309709, 2.51327412],
+                                        [3.14159265, 4.08407045],
+                                        [4.08407045, 2.26194671],
+                                        [4.46106157, 2.38761042]])
+        self.label_train = np.asarray([0, 0, 1, 1])
+
+        self.sample_test = np.asarray([[3.83274304, 2.45044227],
+                                       [3.89557489, 0.31415927]])
+        self.label_test = np.asarray([0, 1])
+
+        self.ref_kernel_train = {
+            'qasm': np.array([[1., 0.84375, 0.12109375, 0.35351562],
+                              [0.84375, 1., 0.1171875, 0.44921875],
+                              [0.12109375, 0.1171875, 1., 0.65136719],
+                              [0.35351562, 0.44921875, 0.65136719, 1.]]),
+            'statevector': np.array([[1., 0.8534228, 0.12267747, 0.36334379],
+                                     [0.8534228, 1., 0.11529017, 0.45246347],
+                                     [0.12267747, 0.11529017, 1., 0.67137258],
+                                     [0.36334379, 0.45246347, 0.67137258, 1.]]),
+            'qasm_sample': np.array([[1., 0.9, 0.2, 0.4],
+                                     [0.9, 1., 0.2, 0.7],
+                                     [0.2, 0.2, 1., 0.9],
+                                     [0.4, 0.7, 0.9, 1.]]),
+            'qasm_sample_psd': np.array([[1.01098001, 0.88069042, 0.18458327, 0.42079597],
+                                         [0.88069042, 1.03395806, 0.22711205, 0.66342794],
+                                         [0.18458327, 0.22711205, 1.02164621, 0.87080094],
+                                         [0.42079597, 0.66342794, 0.87080094, 1.03938728]])
+        }
+
+        self.ref_kernel_test = {
+            'qasm': np.array([[0.14453125, 0.328125],
+                              [0.17675781, 0.36425781],
+                              [0.47363281, 0.02246094],
+                              [0.14648438, 0.16308594]]),
+            'statevector': np.array([[0.1443953, 0.33041779],
+                                     [0.18170069, 0.37663733],
+                                     [0.47479649, 0.02115561],
+                                     [0.14691763, 0.16106199]])
+        }
+
+    def test_qasm_symmetric(self):
+        """ Test symmetric matrix evaluation using qasm simulator """
+        qkclass = QuantumKernel(feature_map=self.feature_map,
+                                quantum_instance=self.qasm_simulator)
+
+        kernel = qkclass.evaluate(x_vec=self.sample_train)
+
+        np.testing.assert_array_almost_equal(kernel,
+                                             self.ref_kernel_train['qasm'],
+                                             decimal=1)
```

Comment:
Does the test also run for `decimal` > 1? 1 seems pretty low 🙂

Reply:
I have rewritten all the test cases to use [np.testing.assert_allclose](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html), with a tolerance of 1e-4.